src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2507,6 +2488,28 @@ ConnectionStateMachine(WorkerSession *session)
 }
 
 
+/*
+ * HandleMultiConnectionSuccess logs the established connection and updates connection's state.
+ */
+static void
+HandleMultiConnectionSuccess(WorkerPool *workerPool, MultiConnection *connection,
+							 WorkerSession *session)
```

Comment:
I think you could reduce this to just the session, and obtain the connection and pool from there.